types/src/transaction.rs

```diff
@@ -1193,3 +1194,18 @@ impl IntoProto for TransactionListWithProof {
         out
     }
 }
+
+/// `ExecutorTransaction` will be the transaction type used internally in the libra node to
+/// represent the transaction to be processed and persisted.
+pub enum ExecutorTransaction {
+    /// Transaction submitted by the user. e.g: P2P payment transaction, publishing module
+    /// transaction, etc.
+    UserTransaction(SignedTransaction),
+
+    /// Transaction that applies a WriteSet to the current storage. This should be used for ONLY for
+    /// genesis right now.
+    WriteSet(WriteSet),
```

Comment:
The `TransactionPayload` inside `RawTransaction` already carries a WriteSet, I'm a little confused what's the difference here?

Reply:
Good question! The plan is we will remove that variant once we are able to persist such item in the storage (see last step of #1173 )